apis/src/websockets/api/schedules/handler.rs

```diff
@@ -0,0 +1,297 @@
+use std::collections::HashMap;
+
+use crate::{
+    common::{ScheduleAction, ScheduleUpdate, ServerMessage},
+    responses::ScheduleResponse,
+    websockets::internal_server_message::{InternalServerMessage, MessageDestination},
+};
+use anyhow::Result;
+use chrono::{DateTime, Utc};
+use db_lib::{
+    get_conn,
+    models::{Game, Tournament},
+    schema::{
+        game_schedules::{self},
+        games::{self},
+    },
+    DbPool,
+};
+
+use diesel::{BoolExpressionMethods, ExpressionMethods, QueryDsl};
+use diesel_async::{scoped_futures::ScopedFutureExt, AsyncConnection, RunQueryDsl};
+use hive_lib::GameError;
+use shared_types::GameId;
+use uuid::Uuid;
+
+pub struct ScheduleHandler {
+    pool: DbPool,
+    user_id: Uuid,
+    action: ScheduleAction,
+}
+
+impl ScheduleHandler {
+    pub async fn new(pool: &DbPool, user_id: Uuid, action: ScheduleAction) -> Result<Self> {
+        Ok(Self {
+            pool: pool.clone(),
+            user_id,
+            action,
+        })
+    }
+
+    pub async fn handle(&self) -> Result<Vec<InternalServerMessage>> {
+        let mut messages = Vec::new();
+        let mut conn = get_conn(&self.pool).await?;
+        match &self.action {
+            ScheduleAction::Propose(date, id) => {
+                let opponent_id = conn
+                    .transaction::<_, anyhow::Error, _>(move |tc| {
+                        async move {
+                            let game = Game::find_by_game_id(id, tc).await?;
+                            self.ensure_user_is_player(&game)?;
+                            diesel::insert_into(game_schedules::table)
```

Comment:
Please move this functionality into its respective model. Doing it here mixes concerns and doesn't make the model reusable. 
Same for the other places.